src/include/migraphx/program.hpp

```diff
@@ -76,7 +76,8 @@ struct program
     void from_value(const value& v);
 
     void debug_print() const;
-    void debug_print(instruction_ref ins) const;
+    void debug_print(instruction_ref ins,
+                     const std::unordered_map<instruction_ref, std::string>& ins_names) const;
```

Comment:
We should not add a map as a parameter as it makes it very difficult(if not impossible) to call from a debugger.

Reply:
@pfultz2 Maybe we can also keep the current function debug_print(instruction_ref ins) for debugging purpose. The reason I made this change is the current function needs to go through the whole program each time an ins is printed. This make the trace_eval=1 very slow, particularly for large model.